FROM ryaneggz/8-add-langchain-mcp-adapter-example TO hwchase17/deepagents@master (#15)
hwchase17/deepagents: README.md

Sub agents are useful for ["context quarantine"](https://www.dbreunig.com/2025/06/26/how-to-fix-your-context.html#context-quarantine) (to help not pollute the overall context of the main agent)
as well as custom instructions.

> ## Adding Langchain MCP Adapter

Reviewer: ```suggestion
## MCP
```

> The `deepagents` library can be ran with MCP tools. This can be extended with the [Langchain MCP Adapter](https://github.com/langchain-ai/langchain-mcp-adapters?tab=readme-ov-file#langchain-mcp-adapters).

Reviewer: ```suggestion
The `deepagents` library can be ran with MCP tools. This can be achieved by using the [Langchain MCP Adapter library](https://github.com/langchain-ai/langchain-mcp-adapters).
```

> (To run the example below, will need to `pip install langchain-mcp-adapter`)

Reviewer: ```suggestion
(To run the example below, will need to `pip install langchain-mcp-adapters`)
```

```python
import asyncio
from langchain_mcp_adapters.client import MultiServerMCPClient
from deepagents import create_deep_agent

async def main():
    # Collect MCP tools
    mcp_client = MultiServerMCPClient(...)
    mcp_tools = await mcp_client.get_tools()

    # Create agent

>     agent = create_deep_agent(
>         tools=mcp_tools,
>         instructions=research_instructions,
>         subagents=[critique_sub_agent, research_sub_agent],
>     ).with_config({"recursion_limit": 1000})

Reviewer: ```suggestion
    agent = create_deep_agent(tools=mcp_tools, ....)
```
